File: android/src/newarch/com/reactnativecommunity/imageeditor/ImageEditorModule.java
package com.reactnativecommunity.imageeditor;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.annotations.ReactModule;

@ReactModule(name = ImageEditorModule.NAME)
public class ImageEditorModule extends NativeRNCImageEditorSpec {
  private ImageEditorModuleImpl moduleImpl;

  ImageEditorModule(ReactApplicationContext context) {
    super(context);
    moduleImpl = new ImageEditorModuleImpl(context);
  }

  public static final String NAME = ImageEditorModuleImpl.NAME;

  @Override
  @NonNull
  public String getName() {
    return ImageEditorModuleImpl.NAME;
  }

  @Override
  public void onCatalystInstanceDestroy() {
    moduleImpl.onCatalystInstanceDestroy();
  }

  @Override
  public void cropImage(String uri, ReadableMap cropData, Promise promise) {
    moduleImpl.cropImage(uri, cropData, promise);
  }
}

File: android/src/newarch/com/reactnativecommunity/imageeditor/ImageEditorModule.kt
package com.reactnativecommunity.imageeditor

import com.facebook.imagepipeline.core.ImagePipeline
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.views.image.ReactCallerContextFactory

@ReactModule(name = ImageEditorModule.NAME)
class ImageEditorModule : NativeRNCImageEditorSpec {
    private val moduleImpl: ImageEditorModuleImpl

    constructor(reactContext: ReactApplicationContext) : super(reactContext) {
        moduleImpl = ImageEditorModuleImpl(reactContext, this, null, null)
    }

    constructor(reactContext: ReactApplicationContext, callerContext: Any?) : super(reactContext) {
        moduleImpl = ImageEditorModuleImpl(reactContext, callerContext, null, null)
    }

    constructor(
        reactContext: ReactApplicationContext,
        imagePipeline: ImagePipeline?,
        callerContextFactory: ReactCallerContextFactory?
    ) : super(reactContext) {
        moduleImpl = ImageEditorModuleImpl(reactContext, null, callerContextFactory, imagePipeline)
    }

    override fun getName(): String {
        return ImageEditorModuleImpl.NAME
    }

    override fun invalidate() {
        moduleImpl.invalidate()
        super.invalidate()
    }

    override fun cropImage(uri: String, cropData: ReadableMap, promise: Promise) {
        moduleImpl.cropImage(uri, cropData, promise)
    }

    companion object {
        const val NAME = ImageEditorModuleImpl.NAME
    }
}
